localstack/services/awslambda/invocation/version_manager.py

```diff
@@ -261,9 +363,27 @@ def invocation_loop(self) -> None:
                     )
                     return
                 LOG.debug("Got invocation event %s in loop", queued_invocation.invocation_id)
+                # Assumption: Synchronous invoke should never end up in the invocation queue because we catch it earlier
```

Comment:
We discussed it, but this assumption is probably incorrect, as some synchronous invocations could still be left in the queue.